contrib/windows/GLPI-AgentMonitor/GLPI-AgentMonitor.cpp

```diff
@@ -0,0 +1,688 @@
+    DWORD imageSize = SizeofResource(hInstance, hResource);
+    if (!imageSize)
+        return false;
+    HGLOBAL tempRes = LoadResource(hInstance, hResource);
+    if (!tempRes)
+        return false;
+    const void* pResourceData = LockResource(tempRes);
+    if (!pResourceData)
+        return false;
+
+    HGLOBAL m_hBuffer = GlobalAlloc(GMEM_MOVEABLE, imageSize);
+    if (m_hBuffer) {
+        void* pBuffer = GlobalLock(m_hBuffer);
+        if (pBuffer) {
+            CopyMemory(pBuffer, pResourceData, imageSize);
+            IStream* pStream = NULL;
+            if (CreateStreamOnHGlobal(m_hBuffer, FALSE, &pStream) == S_OK) {
+                m_pBitmap = Gdiplus::Bitmap::FromStream(pStream);
+                pStream->Release();
+                if (m_pBitmap) {
+                    if (m_pBitmap->GetLastStatus() == Gdiplus::Ok) {
+                        m_pBitmap->GetHBITMAP(0, bitmap);
+                        return true;
+                    }
+                    delete m_pBitmap;
+                    m_pBitmap = NULL;
+                }
+            }
+            m_pBitmap = NULL;
+            GlobalUnlock(m_hBuffer);
+        }
+        GlobalFree(m_hBuffer);
+        m_hBuffer = NULL;
+    }
+    return false;
+}
+
+// Shows a window and force it to appear over all others
+VOID ShowWindowFront(HWND hWnd, int nCmdShow) 
+{
+    ShowWindow(hWnd, nCmdShow);
+    HWND hCurWnd = GetForegroundWindow();
+    DWORD dwMyID = GetCurrentThreadId();
+    DWORD dwCurID = GetWindowThreadProcessId(hCurWnd, NULL);
+    AttachThreadInput(dwCurID, dwMyID, TRUE);
+    SetWindowPos(hWnd, HWND_TOPMOST, 0, 0, 0, 0, SWP_NOSIZE | SWP_NOMOVE);
+    SetWindowPos(hWnd, HWND_NOTOPMOST, 0, 0, 0, 0, SWP_SHOWWINDOW | SWP_NOSIZE | SWP_NOMOVE);
+    SetForegroundWindow(hWnd);
+    SetFocus(hWnd);
+    SetActiveWindow(hWnd);
+    AttachThreadInput(dwCurID, dwMyID, FALSE);
+}
+
+// Requests GLPI Agent status via HTTP and stores it on a wide-char string
+VOID GetAgentStatus(HWND hWnd, LPWSTR szAgStatus, DWORD dwAgStatusLen)
+{
+    DWORD dwSize = 0;
+    DWORD dwDownloaded;
+    vector<BYTE> responseBody;
+
+    LoadString(hInst, IDS_ERR_NOTRESPONDING, szAgStatus, dwAgStatusLen);
+
+    if (running)
+    {
+        HINTERNET hReq = WinHttpOpenRequest(hConn, L"GET", L"/status", NULL, WINHTTP_NO_REFERER, WINHTTP_DEFAULT_ACCEPT_TYPES,
+            WINHTTP_FLAG_BYPASS_PROXY_CACHE);
+
+        WinHttpSendRequest(hReq, WINHTTP_NO_ADDITIONAL_HEADERS, NULL, WINHTTP_NO_REQUEST_DATA, NULL, NULL, NULL);
+
+        if (WinHttpReceiveResponse(hReq, NULL)) {
+            do {
+                dwSize = 0;
+
+                if (!WinHttpQueryDataAvailable(hReq, &dwSize))
+                    break;
+                if (dwSize == 0)
+                    break;
+
+                size_t dwOffset = responseBody.size();
+                responseBody.resize(dwOffset + dwSize);
+
+                if(!WinHttpReadData(hReq, &responseBody[dwOffset], dwSize, &dwDownloaded))
```

Comment:
I see just a readability cosmetic issue here:
```suggestion
                if (!WinHttpReadData(hReq, &responseBody[dwOffset], dwSize, &dwDownloaded))
```